sys/linux/dev_sequencer.txt

```diff
@@ -0,0 +1,173 @@
+ioctl$SNDCTL_SEQ_CTRLRATE(fd fd_seq, cmd const[SNDCTL_SEQ_CTRLRATE], carg ptr[inout, int32])
+ioctl$SNDCTL_SEQ_PANIC(fd fd_seq, cmd const[SNDCTL_SEQ_PANIC])
+ioctl$SNDCTL_SEQ_SYNC(fd fd_seq, cmd const[SNDCTL_SEQ_SYNC])
+ioctl$SNDCTL_SEQ_RESET(fd fd_seq, cmd const[SNDCTL_SEQ_RESET])
+ioctl$SNDCTL_SEQ_TESTMIDI(fd fd_seq, cmd const[SNDCTL_SEQ_TESTMIDI], carg ptr[in, int32])
+ioctl$SNDCTL_SEQ_GETINCOUNT(fd fd_seq, cmd const[SNDCTL_SEQ_GETINCOUNT], carg ptr[out, int32])
+ioctl$SNDCTL_SEQ_GETOUTCOUNT(fd fd_seq, cmd const[SNDCTL_SEQ_GETOUTCOUNT], carg ptr[out, int32])
+ioctl$SNDCTL_SEQ_GETTIME(fd fd_seq, cmd const[SNDCTL_SEQ_GETTIME], carg ptr[out, int32])
+ioctl$SNDCTL_SEQ_RESETSAMPLES(fd fd_seq, cmd const[SNDCTL_SEQ_RESETSAMPLES], carg ptr[in, int32])
+ioctl$SNDCTL_SEQ_NRSYNTHS(fd fd_seq, cmd const[SNDCTL_SEQ_NRSYNTHS], carg ptr[out, int32])
+ioctl$SNDCTL_SEQ_NRMIDIS(fd fd_seq, cmd const[SNDCTL_SEQ_NRMIDIS], carg ptr[out, int32])
+ioctl$SNDCTL_SYNTH_MEMAVL(fd fd_seq, cmd const[SNDCTL_SYNTH_MEMAVL], carg ptr[inout, int32])
+ioctl$SNDCTL_FM_4OP_ENABLE(fd fd_seq, cmd const[SNDCTL_FM_4OP_ENABLE], carg ptr[in, int32])
+ioctl$SNDCTL_SYNTH_INFO(fd fd_seq, cmd const[SNDCTL_SYNTH_INFO], carg ptr[inout, synth_info])
+ioctl$SNDCTL_SYNTH_ID(fd fd_seq, cmd const[SNDCTL_SYNTH_ID], carg ptr[inout, synth_info])
+ioctl$SNDCTL_SEQ_OUTOFBAND(fd fd_seq, cmd const[SNDCTL_SEQ_OUTOFBAND], carg ptr[in, evrec])
+ioctl$SNDCTL_MIDI_INFO(fd fd_seq, cmd const[SNDCTL_MIDI_INFO], carg ptr[inout, midi_info])
+ioctl$SNDCTL_SEQ_THRESHOLD(fd fd_seq, cmd const[SNDCTL_SEQ_THRESHOLD], carg ptr[in, int32])
+ioctl$SNDCTL_MIDI_PRETIME(fd fd_seq, cmd const[SNDCTL_MIDI_PRETIME], carg ptr[inout, int32])
+ioctl$SNDCTL_FM_LOAD_INSTR(fd fd_seq, cmd const[SNDCTL_FM_LOAD_INSTR])
+
+synth_info {
+	name		array[int8, 30]
+	device		int32
+	synth_type	flags[synth_type_flags, int8]
+	synth_subtype	flags[synth_subtype_flags, int32]
+	perc_mode	int32
+	nr_voices	int32
+	nr_drums	int32
+	instr_bank_size	int32
+	capabilities	flags[capabilities_flags, int64]
+	dummies		array[int32, 19]
+}
+
+midi_info {
+	name		array[int8, 30]
+	device		int32
+	capabilities	int32
+	dev_type	int32
+	dummies		array[int32, 18]
+}
+
+evrec_write [
+	s	evrec_short
+	n	evrec_note
+	l	evrec_long
+	v	evrec_voice
+	t	evrec_timer
+	e	evrec_extended
+	x	evrec_sysex
+	echo	int32
+	ch	array[int8, 8]
+	c	evrec_c
+	gen	evrec_gen
+] [varlen]
+
+evrec [
+	s	evrec_short
+	n	evrec_note
+	l	evrec_long
+	v	evrec_voice
+	t	evrec_timer
+	e	evrec_extended
+	x	evrec_sysex
+	echo	int32
+	ch	array[int8, 8]
+	c	evrec_c
+	gen	evrec_gen
+]
+
+#This corresponds to general cases which don't do anything specific with evrec
+evrec_gen {
+	code	flags[evrec_general, int8]
+}
+
+evrec_c {
+	code	flags[evrec_c_code, int8]
+	dev	int8[0:15]
+	arr	array[int8, 6]
+}
+
+evrec_short {
+	code	const[SEQ_MIDIPUTC, int8]
+	parm1	int8
+	dev	int8[0:15]
+	parm2	int8
+}
+
+evrec_note {
+	code	flags[evrec_note_code, int8]
+	chn	int8
+	note	int8
```

Comment:
There are special values for this field, see `note_on_event`. Let's turn this into a union like `evrec_extended_p1` (see below).